src/external/matchers/deepeval.ts

```diff
@@ -0,0 +1,125 @@
+// These metrics are ported from DeepEval.
+// https://docs.confident-ai.com/docs/metrics-conversation-relevancy. See APACHE_LICENSE for license.
+import dedent from 'dedent';
+import { getAndCheckProvider, fail } from '../../matchers';
+import { getDefaultProviders } from '../../providers/defaults';
+import type { GradingConfig, GradingResult } from '../../types';
+import invariant from '../../util/invariant';
+import { getNunjucksEngine } from '../../util/templates';
+
+const nunjucks = getNunjucksEngine(undefined, false, true);
+
+export interface Message {
```

Comment:
Are you absolutely sure you need to define this versus using something already defined?